### PR TITLE
japi: cleanup and add getbuilds function

### PR DESCRIPTION
**Pull request by bmwiedemann**

### CODE REVIEW

File: scripts/jenkins/japi [hunk 1]
```diff
@@ -256,15 +261,22 @@ sub setdescription
   system("curl", "--silent", "-H", $CRUMB, "${jurlbase}/job/${build}/submitDescription?description=$descr");
 }
 
+sub getbuilds
+{
+  my $jobname=shift;
+  $jobname or die;
+  my $list=XMLin(curl("${jurlbase}/job/${jobname}/api/xml?xpath=//number\\&wrapper=builds"), ForceArray => 1);
```
Comment (jdsn) [inline scripts/jenkins/japi]:
```
XML::LibXML->load_xml(
      string => curl(...
```

**Comment by jdsn** [comment]
I'd like to prevent the use if xml-simple. It is quite easy to do the same with libxml. Here is my proposal - feel free to squash it into your commits:
https://github.com/jdsn/SUSE-Cloud--automation/commits/jdsn-pr/354

File: scripts/jenkins/japi [hunk 2]
```diff
@@ -205,12 +208,12 @@ sub push_job($)
...
-  open(my $pipe, "-|", "curl", "--silent", "-H", $CRUMB, "-X", "$METHOD", "--data-binary", "\@${JURL}/${jobname}.xml", $pushurl);
-  if (has_error($pipe)) {
+  my $result=curl("-X", "$METHOD", "--data-binary", "\@${JURL}/${jobname}.xml", $pushurl);
+  if (has_error($result)) {
     print "  updating failed .. ";
     # create
-    open(my $pipe, "-|", "curl", "--silent", "-H", $CRUMB, "-X", "$METHOD", "-H", "Content-Type: text/xml", "--data-binary", "\@${JURL}/${jobname}.xml", "${jurlbase}/createItem?name=${jobname}");
-    if ( has_error($pipe) ) {
+    $result=curl("-X", "$METHOD", "-H", "Content-Type: text/xml", "--data-binary", "\@${JURL}/${jobname}.xml", "${jurlbase}/createItem?name=${jobname}");
+    if (has_error($result)) {
```
Comment (bmwiedemann) [inline scripts/jenkins/japi]: this part needs some testing

Comment (jdsn) [inline scripts/jenkins/japi]: tested and works as expected

**Pull request merged**